File: problems/3171_find_subarray_with_bitwise_or_closest_to_k/Solution.java
import java.util.*;

class Solution {
    public int minimumDifference(int[] nums, int k) {
        int res = Integer.MAX_VALUE;
        Set<Integer> curr = new HashSet<>();
        for (int x : nums) {
            Set<Integer> nxt = new HashSet<>();
            nxt.add(x);
            for (int v : curr) nxt.add(v & x);
            for (int v : nxt) res = Math.min(res, Math.abs(v - k));
            curr = nxt;
        }
        return res;
    }
}

File: problems/3171_find_subarray_with_bitwise_or_closest_to_k/solution.py
from typing import List


class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        res = float('inf')
        curr = set()
        for x in nums:
            nxt = set()
            nxt.add(x)
            for v in curr:
                nxt.add(v & x)
            for v in nxt:
                res = min(res, abs(v - k))
            curr = nxt
        return res
